packages/blog-starter-kit/themes/personal/pages/[slug].tsx

```diff
@@ -191,63 +189,45 @@ export const getStaticProps: GetStaticProps<Props, Params> = async ({ params })
 	if (!params) {
 		throw new Error('No params');
 	}
-	const data = await request<SinglePostByPublicationQuery, SinglePostByPublicationQueryVariables>(
-		process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT,
-		SinglePostByPublicationDocument,
-		{
-			host: process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST,
-			slug: params.slug,
-		},
-	);
 
-	// Extract the post data from the GraphQL response
-	const publication = data.publication;
-	if (!publication) {
+	const endpoint = process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT;
+	const host = process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST;
+	const slug = params.slug;
+
+	const postData = await request(endpoint, SinglePostByPublicationDocument, { host, slug });
+
+	if (postData.publication?.post) {
 		return {
-			notFound: true,
+			props: {
+				type: 'post',
+				post: postData.publication.post,
+				publication: postData.publication,
+			},
 			revalidate: 1,
 		};
 	}
-	const post = publication.post;
-	if (!post) {
-		const staticPageData = await request<PageByPublicationQuery, PageByPublicationQueryVariables>(
-			process.env.NEXT_PUBLIC_HASHNODE_GQL_ENDPOINT,
-			PageByPublicationDocument,
-			{
-				host: process.env.NEXT_PUBLIC_HASHNODE_PUBLICATION_HOST,
-				slug: params.slug,
-			},
-		);
 
-		const page = staticPageData.publication?.staticPage;
-		if (!page) {
-			return {
-				notFound: true,
-				revalidate: 1,
-			};
-		}
+	const pageData = await request(endpoint, PageByPublicationDocument, { host, slug });
+
+	if (pageData.publication?.staticPage) {
 		return {
 			props: {
-				post: null,
-				page,
-				publication,
+				type: 'page',
+				page: pageData.publication.staticPage,
+				publication: pageData.publication,
 			},
 			revalidate: 1,
 		};
 	}
 
 	return {
-		props: {
-			post,
-			page: null,
-			publication,
-		},
+		notFound: true,
 		revalidate: 1,
 	};
 };
 
 export async function getStaticPaths() {
-	const data = await request<SlugPostsByPublicationQuery, SlugPostsByPublicationQueryVariables>(
```

Comment:
@iamandrewluca Encountering an error locally while executing pnpm dev for changes in this branch across all 3 themes (hashnode, personal & enterprise).
```
[next] TypeError: Cannot read properties of undefined (reading 'document')
[next]     at parseRequestExtendedArgs (/Users/harsh/Desktop/hashnode-os/forked_starter-kit/starter-kit/node_modules/.pnpm/graphql-request@6.1.0_graphql@16.8.1/node_modules/graphql-request/build/cjs/parseArgs.js:38:25)
[next]     at request (/Users/harsh/Desktop/hashnode-os/forked_starter-kit/starter-kit/node_modules/.pnpm/graphql-request@6.1.0_graphql@16.8.1/node_modules/graphql-request/build/cjs/index.js:330:72)
```
The functionality is operating without issues in the main branch. Could we please investigate this matter?

Reply:
Hey, @Harsh062! Please check [README Running Locally](https://github.com/Hashnode/starter-kit#running-locally)

Reply:
Thanks @iamandrewluca . I did not have the right ENV properties configured. All 3 themes are working fine after adding the appropriate properties in the env variables file.